src/core/2d_primitives.js

```diff
@@ -13,6 +13,10 @@ var constants = require('./constants');
 var canvas = require('./canvas');
 require('./error_helpers');
 
+// Make mode adjust functions available for testing.
+p5.prototype._modeAdjust = canvas.modeAdjust;
+p5.prototype._arcModeAdjust = canvas.arcModeAdjust;
```

Comment:
is this just being attached to p5.prototype for testing purposes? is there another way to do this? I'd rather avoid adding anything to the prototype that isn't required for functioning of the library, just to avoid bloat of the object.